recognito/src/main/java/com/bitsinharmony/recognito/Recognito.java

```diff
@@ -212,6 +212,55 @@ public VoicePrint createVoicePrint(K userKey, File voiceSampleFile)
 
         return createVoicePrint(userKey, audioSample);
     }
+	
+		/**
+     * Add a voiceprint object. 
+     * <p>
+     * Threading : this method is synchronized to prevent inadvertently erasing an existing user key
+     * </p>
+     * @param userKey the user key associated with this voice print
+     * @param voiceSample the voice print object
+     * @return the voice print itself
+     */
+	
+	public synchronized VoicePrint addVoicePrint(K userKey, VoicePrint voicePrint) {
+		synchronized (this) {
+            if (!universalModelWasSetByUser.get()) {
+                if (universalModel == null) {
+                    universalModel = new VoicePrint(voicePrint);
+                } else {
+                    universalModel.merge(voicePrint.getFeatures());
+                }
+            }
+        }
+		store.put(userKey, voicePrint);
+        return voicePrint;
+	}
+
+    /**
+     * Add a voiceprint based on a current feature vector. 
+     * <p>
+     * Threading : this method is synchronized to prevent inadvertently erasing an existing user key
+     * </p>
+     * @param userKey the user key associated with this voice print
+     * @param features the feature array
+     * @return the voice print extracted from the given features
+     */
+	
+	public synchronized VoicePrint addVoicePrint(K userKey, double[] features) {
```

Comment:
How features are eventually stored in a VoicePrint should remain private API and the user has no reason to depend on it.
A more advanced version of VoicePrint would most likely involve multiple dimensions to the VoicePrint and not just an array of doubles.